### ISSUE TITLE
Crashing when requesting certificate

### ISSUE THREAD

**Issue by durnezj** [infinityofspace/certbot_dns_duckdns#10]
Hi when I request a certificate for my duckDNS domain (normal and wildcard), I get the following error and the application fails to generate a certificate.

```
...
Calling registered functions
Cleaning up challenges
Encountered exception during recovery: certbot.errors.PluginError
Exiting abnormally:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/certbot_dns_duckdns/cert/client.py", line 56, in _perform
    self._get_duckdns_client().set_txt_record(domain, validation)
  File "/usr/local/lib/python3.7/dist-packages/certbot_dns_duckdns/duckdns/client.py", line 49, in set_txt_record
    assert VALID_DUCKDNS_DOMAIN_REGEX.match(domain)
AssertionError

```

Do you have any ideas what might cause this? If you need more info from my end let me know.
Thanks in advance!

**Comment by infinityofspace** [comment]
Can you please specify the exact command of certbot and the versions of certbot and certbot_dns_duckdns you used. **Make sure to anonymize the DuckDNS token and your DuckDNS subdomain.**
Because the domain structure is important for troubleshooting in this case, you should keep the domain structure when anonymizing. As an example of the preservation of the domain structure would be: Your real domain `test1.test2.cloud.infinityofspace.duckdns.org` should become `sub1.sub2.sub3.example.duckdns.org`.

The error means that the domain you specified does not match the two pattern `example.duckdns.org` or `*.example.duckdns.org`. Are you maybe trying to create a certificat for the subsubdomain like `test.example.duckdns.org`?
Unfortunately, support for additional subsubdomains (eg. `test.example.duckdns.org`) is not implemented.
The support for the arbitrary number of additional subsubdomains of the DuckDNS subdomain should be added for the next release v0.5.

**Comment by durnezj** [comment]
Thank for the fast reply!
I've had some time today to dig into the code and was able to fix my issue.
My subdomains on duckdns are constructed like servername-location.duckdns.org.
However, the `assert VALID_DUCKDNS_DOMAIN_REGEX.match(domain)` in _client.py_ caused the certifcate generation to fail because the regex string does not allow hyphens, while this is an allowed character in URLs.
I've proposed a change in a [PR](https://github.com/infinityofspace/certbot_dns_duckdns/pull/11), that fixed the issue for my configuration. I made the PR to inform you about how my problem has been solved, please take a look at it and if you approve it as a general and acceptable solution, you can merge it.

**Comment by infinityofspace** [comment]
Thanks for your troubleshooting effort, I'll take a look at your PR.

**Comment by infinityofspace** [comment]
closed with #11
